src/v/kafka/server/connection_context.h

```diff
@@ -120,14 +137,65 @@ class connection_context final
     using sequence_id = named_type<uint64_t, struct kafka_protocol_sequence>;
     using map_t = absl::flat_hash_map<sequence_id, response_ptr>;
 
+    class ctx_log {
+    public:
+        ctx_log(const ss::net::inet_address& addr, uint16_t port)
+          : _client_addr(addr)
+          , _client_port(port) {}
+
+        template<typename... Args>
+        void error(const char* format, Args&&... args) {
+            log(ss::log_level::error, format, std::forward<Args>(args)...);
+        }
+        template<typename... Args>
+        void warn(const char* format, Args&&... args) {
+            log(ss::log_level::warn, format, std::forward<Args>(args)...);
+        }
+
+        template<typename... Args>
+        void info(const char* format, Args&&... args) {
+            log(ss::log_level::info, format, std::forward<Args>(args)...);
+        }
+
+        template<typename... Args>
+        void debug(const char* format, Args&&... args) {
+            log(ss::log_level::debug, format, std::forward<Args>(args)...);
+        }
+
+        template<typename... Args>
+        void trace(const char* format, Args&&... args) {
+            log(ss::log_level::trace, format, std::forward<Args>(args)...);
+        }
+
+        template<typename... Args>
+        void log(ss::log_level lvl, const char* format, Args&&... args) {
+            if (klog.is_enabled(lvl)) {
+                auto line_fmt = ss::sstring("{}:{} failed authorization - ")
+                                + format;
+                klog.log(
+                  lvl,
+                  line_fmt.c_str(),
+                  _client_addr,
+                  _client_port,
+                  std::forward<Args>(args)...);
+            }
+        }
+
+    private:
+        const ss::net::inet_address& _client_addr;
```

Comment:
can you make a comment of who owns the const-ref member?

Reply:
IDK why github didn't mark this as updated, but I did put the comment line in.